packages/pigeon/lib/java_generator.dart

```diff
@@ -449,7 +449,9 @@ class JavaGenerator extends StructuredGenerator<JavaOptions> {
 
       for (final Method func in api.methods) {
         final String resultType =
-            func.returnType.isNullable ? 'NullableResult' : 'Result';
+            func.returnType.isNullable || func.returnType.isVoid
+                ? 'NullableResult'
+                : 'Result';
```

Comment:
We should probably extract a helper instead of repeating this non-trivial logic three times (and put a comment explaining why void is nullable).